packages/analytics/index.tsx

```diff
@@ -28,9 +28,47 @@ export default function usePageviewAnalytics(): void {
         ),
       })
 
+      // If fathom hasn't loaded, warn the user and shortcircuit
+      if (!window.fathom) {
+        console.warn(
+          '[@hashicorp/platform-analytics] The Fathom script has failed to load.'
+        )
+        return
+      }
+
       // Record a pageview when route changes
       router.events.on('routeChangeComplete', onRouteChangeComplete)
 
+      return () => {
+        router.events.off('routeChangeComplete', onRouteChangeComplete)
+      }
+    } else if (
+      process.env.HASHI_ENV === 'preview' &&
+      process.env.NEXT_PUBLIC_FATHOM_DEV_SITE_ID
+    ) {
+      /**
+       * Send analytics traffic to a test site in Fathom
+       * when running in dev, but only when dev env vars
+       * are set.
+       *
+       * This allows optional testing of new instances
+       * locally before merge, without affecting live
+       * analytics.
```

Comment:
Should also add here, "new instances and tracking client calls," such as to track events via Fathom.trackGoal().